python/raydp/spark/ray_cluster.py

```diff
@@ -88,6 +88,12 @@ def get_spark_session(self,
         else:
             extra_conf[driver_cp_key] = driver_cp
         spark_builder = SparkSession.builder
+        dyn_alloc_key = "spark.dynamicAllocation.enabled"
+        if dyn_alloc_key in extra_conf and extra_conf[dyn_alloc_key] == "true":
+            max_executor_key = "spark.dynamicAllocation.maxExecutors"
+            # set max executors if not set. otherwise spark might request too many actors
+            if max_executor_key not in extra_conf:
+                print("Warn: Consider setting max executor number to match the cluster config.")
```

Comment:
Better to be more specific here. Let's directly call out setting spark.dynamicAllocation.maxExecutors in the message?